### PR TITLE
Update chaos-mesh and the controller runtime version

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Updates the controller runtime to a more recent version and updates the chaos-mesh API definitions.

## Type of change

*Please select one of the options below.*

- Other

## Discussion

-

## Testing

Manual e2e test run.

## Documentation

-

## Follow-up

-

### CODE REVIEW

File: controllers/backup_controller.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func (r *FoundationDBBackupReconciler) updateOrApply(ctx context.Context, backup
 			Status: backup.Status,
 		}
 
-		return r.Status().Patch(ctx, patch, client.Apply, client.FieldOwner("fdb-operator"), client.ForceOwnership)
+		return r.Status().Patch(ctx, patch, client.Apply, client.FieldOwner("fdb-operator")) //, client.ForceOwnership)
```
Comment (sbodagala) [inline controllers/backup_controller.go:200]: I am not sure how this change is related to upgrading chaos-mesh.

File: controllers/cluster_controller.go [hunk 2]
```diff
@@ -220,6 +220,13 @@ func (r *FoundationDBClusterReconciler) SetupWithManager(mgr ctrl.Manager, maxCo
 		return err
 	}
 
+	err = mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Node{}, "metadata.name", func(o client.Object) []string {
+		return []string{o.(*corev1.Node).Name}
```
Comment (sbodagala) [inline controllers/cluster_controller.go:224]: Why are we making this change? Overall, I think I am not following the controller related changes (a summary of why these changes are being made would be nice).

**Review by sbodagala — APPROVED** [review]

File: controllers/backup_controller.go (hunk 1 above)
Comment (johscheuer) [reply controllers/backup_controller.go:200]: Those changes are related to the new controller runtime, there were some changes that made the `client.ForceOwnership` not a correct argument for sub resources.

File: controllers/cluster_controller.go (hunk 2 above)
Comment (johscheuer) [reply controllers/cluster_controller.go:224]: Those where required by the the controller-runtime as they added some checks if all the requested fields have and indexer and when the node taint support was added, this was missed and the new controller-runtime cached this issue.
